substrate/frame/revive/src/lib.rs

```diff
@@ -293,6 +293,13 @@ pub mod pallet {
 		/// This value is usually higher than [`Self::RuntimeMemory`] to account for the fact
 		/// that validators have to hold all storage items in PvF memory.
 		type PVFMemory: Get<u32>;
+
+		/// The [EIP-155](https://eips.ethereum.org/EIPS/eip-155) chain ID.
+		///
+		/// This is a unique identifier assigned to each blockchain network,
+		/// preventing replay attacks.
+		#[pallet::constant]
+		type ChainId: Get<u64>;
```

Comment:
Why not U256 if that is the native format of the ID?

Reply:
People seem to [generally pick them within the bounds of 64bits](https://chainlist.org/). I guess they want to have easily readable for humans? Also there is no `ConstU256` yet. But it this be fine, I hope humanity never needs more than `2**64` blockchains :P

Reply:
Ah I have the same patch in my branch I used u32, there..., both are fine I guess

Reply:
Ok let's stick with u64. As long as we don't need to convert back to u64 it's fine.

Reply:
On the RPC side I'm not sure but for the runtime I don't see where we would need to parse this ever.

Reply:
However even if so, I feel like this would be a fallible conversion anyways.